### ISSUE TITLE
Schema generation fails when multiple interfaces of a type implement the same base interface

### ISSUE THREAD

**Issue by mzhu22** [strawberry-graphql/strawberry#2646]
## Describe the Bug

The following is a valid GraphQL schema, which includes interfaces that implement other interfaces:
```graphql
interface Base {
  id: String
}

interface InterfaceA implements Base {
  id: String
  fieldA: String
}

interface InterfaceB implements Base {
  id: String
  fieldB: String
}

type MyType implements InterfaceA & InterfaceB & Base {
  id: String
  fieldA: String
  fieldB: String
}
```

We implement it in Strawberry like this, using inheritance:
```python
import strawberry

@strawberry.interface
class Base:
    id: str

@strawberry.interface
class InterfaceA(Base):
    id: str
    field_a: str
    
@strawberry.interface
class InterfaceB(Base):
    id: str
    field_b: str

@strawberry.type
class MyType(InterfaceA, InterfaceB):
    id: str
    field_a: str
    field_b: str  
```

However, this results in an error when generating the GraphQL schema:
```
E           ValueError: Invalid Schema. Errors:
E           
E           ❌ Type MyType can only implement Base once.
```

As the generated schema has:
```graphql
type MyType implements InterfaceA & Base & InterfaceB & Base {
  id: String
  fieldA: String
  fieldB: String
}
```
I think the issue is in https://github.com/strawberry-graphql/strawberry/blob/20e202757ada7709e189f2b9b9cef6cb6023d5f5/strawberry/object_type.py#L33

I believe that the function `_get_interfaces` should be deduplicating the list of interfaces it gets by recursively traversing through a type's bases.

## System Information

 - Operating system: MacOS Ventura 13.1 (22C65)
 - Strawberry version (if applicable): 0.164.0

**Comment by skilkis** [comment]
@mzhu22 thank you for the bug report and the suggested location where this might be ground wrong! I'll get on this soon

**Comment by patrick91** [comment]
@mzhu22 this was fixed by @skilkis  in #2647, feel free to reopen if the issue is still there!
